src/simulator_processing/infrared_id_compensation.cpp

```diff
@@ -54,14 +54,19 @@ bool InfraredIdCompensation::setupFromRos(const ros::NodeHandle& nh,
 }
 
 void InfraredIdCompensation::imageCallback(const sensor_msgs::ImagePtr& msg) {
-  cv_bridge::CvImagePtr img = cv_bridge::toCvCopy(msg, msg->encoding);
-  for (int v = 0; v < img->image.rows; v++) {
-    for (int u = 0; u < img->image.cols; u++) {
-      img->image.at<uchar>(v, u) =
-          infrared_compensation_[img->image.at<uchar>(v, u)];
+  try {
```

Comment:
Prefer specific checks to try blocks where possible. E.g. if the issue is that the messages are sometimes empty you can directly check for that. If it fails for various reasons then try-catch is fine. The entire image conversion loop does not need to be in the try block I think? The only thing that could go wrong here is that the data is out of range, which you can statically check for when loading the infrared corrections (i.e. that all values between 0-255 are covered)